pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -113,9 +120,72 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .filter(t -> (t.getInserted().matches("[a-zA-Z/]")))
+                           .subscribe(t -> {
+                               try {
+                                   if (xpathExpressionArea.getText().contains("/")) {
+                                       if (!t.getInserted().equals("/")) {
+                                           autoComplete(xpathExpressionArea.getText().substring(xpathExpressionArea.getText().lastIndexOf("/")));
```

Comment:
This takes all the text after the last `/`, which could contain eg several predicates, with possibly whitespace that you shouldn't strip (they start with `[`). Here's the outline of a better procedure

* if the inserted text contains `/`, cut out the part between the last `/` that's within the inserted text and the end of the inserted text
  * else cut out the part between the last `/` that's before the insertion point (t.getPosition()) and the end of the inserted text
* continue only if that segment is only letters
* compute suggestions from it and update the menu. 
  * if a suggestion is selected, then replace this text by the selected suggestion (but nothing else).

Give a more descriptive name to `t` as well